### PR TITLE
Adds documentation for ClickHouse support

### PR DESCRIPTION
**Pull request by linghengqian**
For #29052.

Changes proposed in this pull request:
  - Adds documentation for ClickHouse support.
  - TL;DR, no developer would like to see a long discussion about ClickHouse in the documentation of GraalVM Native Image integration. It is necessary to separate the independent documentation of ClickHouse.
  - Add additional instructions for `org.testcontainers.jdbc.ContainerDatabaseDriver#killContainers()` in the testcontainers document. Some downstream projects have enabled the parallel execution feature of unit tests in Junit5, but this feature is basically useless for testcontainers-java because the concurrency capability of the Client/Server architecture of Docker Engine may not be that strong.
  - Update docs for hiveserver2 integration. Since #33279 was merged, the `org.apache.commons:commons-text` dependency used by shardingsphere jdbc started to conflict with https://github.com/apache/hive/blob/rel/release-4.0.1/jdbc/pom.xml .

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.
- [x] I have updated the Release Notes of the current development version. For more details, see [Update Release Note](https://shardingsphere.apache.org/community/en/involved/contribute/contributor/)

### CODE REVIEW

File: docs/document/content/user-manual/shardingsphere-jdbc/optional-plugins/clickhouse/_index.en.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+When configuring ShardingSphere to connect to ClickHouse, 
+if ShardingSphere is also configured to use the Key Generate SPI implementation of `SNOWFLAKE`,
+the column type in the ClickHouse real database used by ShardingSphere's Key Generate function should not be set to `UUID`.
+
+Because `com.clickhouse.jdbc.ClickHouseConnection#prepareStatement(String, int)` of `com.clickhouse:clickhouse-jdbc:0.6.3:http`
+Maven module intentionally throws an exception when `autoGeneratedKeys` is `java.sql.Statement.RETURN_GENERATED_KEYS`,
+to prevent ShardingSphere from proxying `com.clickhouse.jdbc.internal.ClickHouseConnectionImpl` normally,
+therefore, if users need to obtain the Key generated by ShardingSphere from the JDBC business code, 
+they need to set `autoGeneratedKeys` to `java.sql.Statement.NO_GENERATED_KEYS`.
```
Comment (linghengqian) [inline docs/document/content/user-manual/shardingsphere-jdbc/optional-plugins/clickhouse/_index.en.md:182]: - I opened https://github.com/ClickHouse/clickhouse-java/issues/1968 for this. Related issues already exist from a few years ago.

**Review by strongduanmu — APPROVED** [review]
